client/analytics/report/categories/index.js

```diff
@@ -55,6 +62,10 @@ export default class CategoriesReport extends Component {
 				: 'category';
 		}
 
+		filters[ 0 ].filters.find(
+			( item ) => item.value === 'compare-categories'
+		).settings.onClick = addCesSurveyTrackForAnalytics;
```

Comment:
I initially handled it then changed it back as per @becdetat 's suggestion at https://github.com/woocommerce/woocommerce-admin/pull/5649#discussion_r525617526. What do you think about the suggestion?

Reply:
That's a very good point. A fast-fail is often much more useful at tracking down things.

My main concern is that if this errors, the whole CategoriesReport won't be rendered, which seems like a pretty big failure from a user standpoint.

Maybe some automated testing should ultimately be added for the report components so that we can be assured that we will quickly notice if these filters get removed or renamed. But, that could be a future PR.

I'm okay with the fast-fail approach for now.

Reply:
@mattsherman That's a good point. I will try to add a test case if time allows 👍